addons/sale_timesheet/models/project.py

```diff
@@ -175,6 +176,44 @@ def _compute_sale_line_id(self):
             ], limit=1)
             project.sale_line_id = sol or project.sale_line_employee_ids.sale_line_id[:1]  # get the first SOL containing in the employee mappings if no sol found in the search
 
+    @api.depends('sale_line_id.product_uom_qty', 'sale_line_id.product_uom')
+    def _compute_allocated_hours(self):
+        sale_orders = self.sale_order_id | self.tasks.sale_order_id
+        sols_read_group = self.env['sale.order.line'].read_group([
+            ('order_id', 'in', sale_orders.ids), ('is_service', '=', True),
+            ('is_downpayment', '=', False), ('product_id.project_template_id', '=', False),
```

Comment:
It seems Github lost my previous comment 🙁 I'll rewrite it here down.

After having a bit more thought of it, I am a bit concerned regarding performance since we are using self.tasks which is a one2many field and that the volume can be quite huge in some big databases. So here are the problems is see:
- the ORM will fetch all the records as it is a one2many. We could solve this by performing a search_read on `project.task` in order to get the sale_order_id for the project_id in self.ids, then create a set to remove the duplicates and create a recordset by performing a browse of the ids in the set and finally add the recordset to self.sale_order_id.
- the amount of ids in sale_orders.ids can be quite large and I am unsure that this will be performant / supported.

That's why I am wondering whether we shouldn't rather use a SQL here. What do you think @xavierbol ?

Reply:
@IT-Ideas , I agree on this as we are using the same logic in `_get_sale_order_lines` and `_get_all_sales_orders` which is a performance killer too so we can refactor those methods(with query or search_read) and also support the extra domain(with default value) on those methods so we can maximize the use of that method.

cc @xavierbol 

Thanks